partiql-ast/src/ast.rs

```diff
@@ -871,6 +871,8 @@ pub struct CustomType {
 #[derive(Clone, Debug, PartialEq)]
 pub struct SymbolPrimitive {
     pub value: String,
+    // E.g. "date"
+    pub dbl_quoted: bool,
```

Comment:
Thought about that but then realized apparently there are two uses for double quotes:
a. To disambiguate from a keyword and an identifier
b. To specify case sensitivity for attribute look-ups

That's why I added `dbl_quoted` for the first use (a). Let me know if you think otherwise.

Reply:
I don't think there's any difference between the two. The need to disambiguate keywords with quotes is so the Lexer knows to emit an Identifier token, not a keyword token (this is why names with `-`, `+`, etc. must also be quoted, so they Lexer doesn't parse the punctuation separately).

Reply:
That's correct, I just did a further probing to check when keyword is quoted will it be considered as case sensitive attribute or not—as from the current Kotlin impl, it indeed gets treated as case sensitive attribute:
```bash
PartiQL> PIVOT t.price AT t."symbol" FROM {'symbol':'tdc', 'price': 31.52} AS t;
==='
{
  'tdc': 31.52
}
---
OK!
PartiQL> PIVOT t.price AT t."symbol" FROM {'Symbol':'tdc', 'price': 31.52} AS t;
==='
{}
---
OK!
```

I'll make the following change:
```rust
#[derive(Clone, Debug, PartialEq)]
pub struct SymbolPrimitive {
    pub value: String,
    pub case: CaseSensitivity,
}
```

Reply:
I just recalled, we also need to cater for `SymbolPrimitives` for string literals (E.g. 'a') in that case it's neither case-sensitive, case-insensitive. So perhaps `Option(CaseSensitivity)` makes more sense.